core/src/main/scala/cr/pulsar/internal/FutureLift.scala

```diff
@@ -51,10 +51,10 @@ object FutureLift {
                 cb(Left(ex))
             }
           }
-          F.delay(f.cancel(true)).void
+          F.delay(Option(F.delay(f.cancel(true)).void))
```

Comment:
This doesn't look good. Is there a way to do it better?

Reply:
You can use `Some` directly but that's it.

I think we can create a `FutureLift` capability so other methods only require this constraint instead of `Async` directly, see this: https://github.com/profunktor/redis4cats/blob/series/1.x/modules/core/src/main/scala/dev/profunktor/redis4cats/effect/FutureLift.scala#L48

Reply:
Actually, we can have the `FutureLift` trait, use this wherever a conversion is needed, and have a default instance for `Async`. The rest of the implementation can now be removed, we can use `Async[F].fromCompletableFuture`